### PR TITLE
Updated eligibility rules for Colorado utility bill help program similar to WeatherizationAssistance pattern.

### PR DESCRIPTION
**Pull request by SonaliBedge**
## Context & Motivation

- Fixes #325
- Related PR: [link if applicable]

## Changes Made

- ...

## Testing

- Migrations to run:
- Configuration updates needed:
- Environment variables/settings to add:
- Manual testing steps:

## Deployment

- Run script:
- Update production config:
- Admin updates needed:
- Notify team/users of:

## Notes for Reviewers

- Known limitations:
- Future considerations:

### CODE REVIEW

File: programs/programs/co/utility_bill_pay/calculator.py [hunk 1]
```diff
@@ -23,6 +46,18 @@ def household_eligible(self, e: Eligibility):
             if any(member.has_benefit(benefit) for member in self.screen.household_members.all()):
                 presumptive_eligible = True
 
+        # income condition
+        counties = counties_from_screen(self.screen)
+        income_limits = []
+        for county in counties:
+            income_limits.append(self.income_limits.fetch()[county][self.screen.household_size - 1])
+        income_limit = min(income_limits)
```
[the remark this answers is not in the document]
Comment (SonaliBedge) [reply programs/programs/co/utility_bill_pay/calculator.py:82]: I made all the suggested changes above, except for the one related to counties, because there is already validation for counties on the frontend. Users can only proceed by entering county information, so I don’t think we need to add county validation here.

File: programs/programs/co/utility_bill_pay/calculator.py [hunk 2]
```diff
@@ -23,6 +46,18 @@ def household_eligible(self, e: Eligibility):
...
+
+        income = int(self.screen.calc_gross_income("yearly", ["all"]))
```
[the remark this answers is not in the document]
Comment (SonaliBedge) [reply programs/programs/co/utility_bill_pay/calculator.py:84]: This is a calc_gross_income function code , which always return float " ```
def calc_gross_income(self, frequency, types, exclude=[]):
        household_members = self.household_members.all()
        gross_income = 0

        for household_member in household_members:
            gross_income += household_member.calc_gross_income(frequency, types, exclude)
        return float(gross_income)
```"

File: programs/programs/co/utility_bill_pay/calculator.py [hunk 3]
```diff
@@ -23,6 +61,31 @@ def household_eligible(self, e: Eligibility):
...
+            if not county_data:
+                continue
+
+            # Validate household_size bounds (1-8)
+            if size_index < 0 or size_index >= len(county_data):
+                continue
+            income_limits.append(county_data[size_index])
+
+        if not income_limits:
+            e.condition(False, messages.income())
```
Comment (jmejia) [inline programs/programs/co/utility_bill_pay/calculator.py:80]: Looks like the messages.income() that's imported requires two args that aren't being passed in. Maybe we can write a test for this edge case?

Comment (SonaliBedge) [reply programs/programs/co/utility_bill_pay/calculator.py]: I have added two args for messages.income() and handled missing counties.

**Comment by SonaliBedge** [comment]
@coderabbitai review

File: programs/programs/co/utility_bill_pay/calculator.py [hunk 4]
```diff
@@ -23,6 +61,33 @@ def household_eligible(self, e: Eligibility):
...
+
+            # Validate household_size bounds (1-8)
+            if size_index < 0 or size_index >= len(county_data):
+                continue
+            income_limits.append(county_data[size_index])
+
+        income = int(self.screen.calc_gross_income("yearly", ["all"]))
+        if not income_limits:
+            e.condition(False, messages.county_not_found())
+            return
```
[the remark this answers is not in the document]
Comment (SonaliBedge) [reply programs/programs/co/utility_bill_pay/calculator.py:84]: I am done with above minor change, please review.

File: programs/programs/co/energy_programs_shared/income_validation.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+               - income_limit: The income limit (or None if not found)
+               - error_detail: Error reason string (or None if successful)
+    """
+    # Get county from screen (handles both direct county selection and zipcode lookup)
+    counties = counties_from_screen(screen)
+    if not counties:
+        return None, "no_county_provided"
+
+    # Get formatted county name
+    county = _get_county_name(screen)
```
Comment (catonph) [inline programs/programs/co/energy_programs_shared/income_validation.py]: So you don't need a function to do do any lookups around the county (plus we wouldn't want to make an assumption that we should use the first county in the list of counties). 

Instead, you just can reference `screen.county`. Check out the `Screen` model in benefits-api/screener/models.py on line 26. It's already a Screen field.

File: programs/programs/co/energy_programs_shared/income_validation.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    limits_by_county = income_limits_cache.fetch()
+    size_index = screen.household_size - 1
+
+    # Check conditions using if/elif (only one condition will be checked)
+    if county not in limits_by_county:
+        return None, "county_not_found"
+    elif limits_by_county.get(county) is None:
+        return None, "county_data_none"
+    elif size_index < 0 or size_index >= len(limits_by_county[county]):
+        return None, "HH_size_out_of_bounds"
```
[the remark this answers is not in the document]
Comment (catonph) [reply programs/programs/co/energy_programs_shared/income_validation.py]: Also we are adding unnecessary complexity by returning the error codes here - just update the eligiblity message in line for each of these conditions

File: programs/programs/messages.py [hunk 7]
```diff
@@ -14,6 +17,20 @@ def income(income, max_income):
     )
 
 
+def income_limit_unknown(reason: str, county: str, size_index: int | None = None):
+    """
+    Log income limit failure to Sentry with a specific reason,
+    but return the same user-facing message.
+    """
+    msg = f"Income limit failure for county={county}. Reason={reason}"
```
Comment (catonph) [inline programs/programs/messages.py]: I think you are looking for this syntax (instead of embedding those fields in the string): 

```
capture_message(
    "Your message here",
    level="error",
    extras={
        "count": county,
        "details": reason,
    }
)
```

**Comment by SonaliBedge** [comment]
Hi @jmejia and @patwey , Yesterday, @catonph and I did some pair programming to make a few changes. Please review them whenever it's convenient for you. Thank you Caton!
